product/roundhouse.tasks/Roundhouse.cs

```diff
@@ -142,12 +153,14 @@ public string RecoveryMode
             set
             {
                 RecoveryMode result;
-                if (!Enum.TryParse(value, true, out result))
+                if (Enum.TryParse(value, true, out result))
                 {
-                    throw new ArgumentOutOfRangeException("value", value, "The value of 'RecoveryMode' must be one of these values: 'NoChange', 'Simple' or 'Full'.");
+                    this.recoveryMode = result;
+                }
+                else
+                {
+                    this.loggingHelper.LogError("The value of 'RecoveryMode' must be one of these values: 'NoChange', 'Simple' or 'Full'. Actual value was '{0}'.", value);

```

Comment:
FWIW, to move forward, at this time I don't want the logginghelper or msbuildtasks referenced. We can take a look at what is set during execute (after everything is loaded) and throw an error at that time like we already have a convention for.